File: app/src/main/java/com/fltry/androidlibs/sdkmodule/retrofit/RetrofitActivity.java
package com.fltry.androidlibs.sdkmodule.retrofit;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.fltry.androidlibs.R;
import com.fltry.androidlibs.sdkmodule.retrofit.base.MVPActivity;
import com.fltry.androidlibs.ui.BaseActivity;

public class RetrofitActivity extends MVPActivity<IResult,LoginPresenter> implements IResult{

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getToolbarTitle().setText("Retrofit+rxandroid+MVP");
    }

    @Override
    protected LoginPresenter createPresenter() {
        return new LoginPresenter();
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_retrofit;
    }


    @Override
    public void showDialog() {

    }

    @Override
    public void dismissDialog() {

    }

    @Override
    public void loginSucceed() {

    }

    @Override
    public void loginFailed() {

    }
}

File: app/src/main/java/com/fltry/androidlibs/sdkmodule/retrofit/RetrofitActivity.kt
package com.fltry.androidlibs.sdkmodule.retrofit

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import butterknife.OnClick
import com.fltry.androidlibs.R
import com.fltry.androidlibs.ui.BaseActivity

class RetrofitActivity : BaseActivity() {
    lateinit var retrofitEt: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        toolbarTitle!!.text = "retrofit框架的使用"

        retrofitEt = findViewById(R.id.retrofit_et)
        retrofitEt.setText("http://10.10.10.107:8004/areas")
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_retrofit
    }

    @OnClick(R.id.retrofit_et)
    fun onClick(v: View) {
        when (v.id) {
            R.id.retrofit_btn -> {
                Log.i("aaa", "adsfasdfsadf")
            }
        }
    }
}
